packages/ec-core/src/distributions/wrappers/owned.rs

```diff
@@ -57,14 +57,15 @@ where
     /// # Errors
     /// - [`EmptySlice`] if an empty collection is passed in, since then no
     ///   element can be selected from there
-    #[allow(clippy::unwrap_used)]
-    #[allow(clippy::missing_panics_doc)]
     pub fn new(collection: T) -> Result<Self, EmptySlice> {
         let num_choices = NonZeroUsize::new(collection.borrow().len()).ok_or(EmptySlice)?;
 
         Ok(Self {
             collection,
-            range: Uniform::new(0, num_choices.get()).unwrap(),
+            // this is technically guaranteed to be Ok() but I figured there is no huge difference
+            // between .unwrap() and .map_err() here and .unwrap() is arguably easier to
+            // get wrong.
+            range: Uniform::new(0, num_choices.get()).map_err(|_| EmptySlice)?,
```

Comment:
yea maybe just remove the comment - Although I also like to mention that the case should never, for any reason, actually be triggered.

Reply:
I do think there should be _some_ sort of comment for the reasons you gave. I just don't think that we want to include the history in the comment since that will make less and less sense down the line.